src/Workspaces/Core/Portable/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.AnalysisData.cs

```diff
@@ -247,20 +240,7 @@ public void SetCurrentBlockAnalysisDataFrom(BasicBlockAnalysisData newBlockAnaly
                 CurrentBlockAnalysisData.SetAnalysisDataFrom(newBlockAnalysisData);
             }
 
-            public void Dispose()
-            {
-                DisposeAllocatedPrivateData();
-                DisposeCoreData();
-            }
-
-            protected virtual void DisposeCoreData()
-            {
-                SymbolsWriteBuilder.Free();
-                SymbolsReadBuilder.Free();
-                LambdaOrLocalFunctionsBeingAnalyzed.Free();
-            }
-
-            private void DisposeAllocatedPrivateData()
+            public virtual void Dispose()
```

Comment:
Using the Dispose pattern recommended by @sharwell here: https://github.com/dotnet/docs/issues/8463#issuecomment-451003282

Reply:
`Dispose()` should be `virtual`, not `abstract`, so override implementations will always call the base implementation.

Reply:
Also, I lean towards `Dispose` going in reverse order of construction: the derived type disposes members first, then calls `base.Dispose()` before returning.

Reply:
Thanks!